packages/api/core/src/util/forge-config.ts

```diff
@@ -123,14 +125,20 @@ export default async (dir: string): Promise<ResolvedForgeConfig> => {
   forgeConfig = forgeConfig || ({} as ForgeConfig);
 
   if (await forgeConfigIsValidFilePath(dir, forgeConfig)) {
+    const forgeConfigPath = path.resolve(dir, forgeConfig as string);
     try {
       // The loaded "config" could potentially be a static forge config, ESM module or async function
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const loaded = require(path.resolve(dir, forgeConfig as string)) as MaybeESM<ForgeConfig | AsyncForgeConfigGenerator>;
+      let loaded;
+      try {
+        loaded = (await dynamicImport(forgeConfigPath)) as MaybeESM<ForgeConfig | AsyncForgeConfigGenerator>;
+      } catch (err) {
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        loaded = require(forgeConfigPath) as MaybeESM<ForgeConfig | AsyncForgeConfigGenerator>;
+      }
```

Comment:
@erikian actually It couldn't be simplified, because require supports custom loader from `interpret` module https://github.com/electron/forge/blob/main/packages/api/core/src/util/forge-config.ts#L5 while import doesn't support it. So it is necessary to have require to support yml, json etc configs.